lib/pure/math.nim

```diff
@@ -77,6 +77,36 @@ when defined(c) or defined(cpp):
       importc: "frexpf", header: "<math.h>".}
   func c_frexp2(x: cdouble, exponent: var cint): cdouble {.
       importc: "frexp", header: "<math.h>".}
+  
+  type
+    div_t {.importc, header: "<stdlib.h>".} = object
+      quot: cint
+      rem: cint
+    ldiv_t {.importc, header: "<stdlib.h>".} = object
+      quot: clong
+      rem: clong
+    lldiv_t {.importc, header: "<stdlib.h>".} = object
+      quot: clonglong
+      rem: clonglong
+  
+  when cint isnot clong:
+    func divmod_c(x, y: cint): div_t {.importc: "div", header: "<stdlib.h>".}
+  when clong isnot clonglong:
+    func divmod_c(x, y: clonglong): lldiv_t {.importc: "lldiv", header: "<stdlib.h>".}
+  func divmod_c(x, y: clong): ldiv_t {.importc: "ldiv", header: "<stdlib.h>".}
+  func divmod*[T: SomeInteger](x, y: T): (T, T) {.inline.} = 
```

Comment:
since this is in `math`, shouldn't it raise Defect on overflow etc?

Reply:
SIGFPE is likely a division by zero - overflow on the other hand happens when doing `high/-1`: https://stackoverflow.com/questions/30394086/integer-division-overflows

both cases would need to be checked for before performing the actual division

Reply:
Well, low/-1 because these are signed integers. That is what I used to test and I get the SIGFPE error. I took a look in the compiler source and I see the implementation of these overflow errors in ccgexprs.nim. I can do something similar to what Nim's `div` does for this instruction if I am on the right track here. Please advise

Reply:
right, forgot that -1 gives a SIGFPE as well - both of these cases have to be done the same way as in ccgexprs, ie check-before. 

you should be able to use `when compileOption("overflowChecks")` to disable them selectively, though this is good to add to the test suite as well.

Re performance, compilers detect div followed by mod and issue the optimal (ie single) cpu instruction regardless (this is an entirely trivial transformation, see https://godbolt.org/z/WvEPhE5c1 for example) - there's no performance reason to call this function really.